webapp/components/search_results_item.jsx

```diff
@@ -70,14 +86,35 @@ export default class SearchResultsItem extends React.Component {
         GlobalActions.emitPostFocusRightHandSideFromSearch(this.props.post, this.props.isMentionSearch);
     }
 
-    flagPost(e) {
-        e.preventDefault();
-        flagPost(this.props.post.id);
+    handleJumpClick() {
+        if (Utils.isMobile()) {
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_SEARCH,
+                results: null
+            });
+
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_SEARCH_TERM,
+                term: null,
+                do_search: false,
+                is_mention_search: false
+            });
+
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_POST_SELECTED,
+                postId: null
+            });
```

Comment:
Can you add an action to do the app dispatches? We should avoid using AppDispatcher directly in components